### DISCUSSION TITLE
Oxford to EDAX Workflow

### DISCUSSION THREAD

**Discussion by MJPavel1** [hyperspy/rosettasciio#214] [General]
Has anyone sorted out a good workflow for taking saved pattern EBSD data from an Oxford system and importing into OIM for reindexing / analysis purposes?

We like OIM for the analysis capability but our microscope with APEX / EDAX is really awful so we would like to collect all our EBSD on the microscope which is run by aztec. Ever since the new update, the file importing from Oxford -> OIM is just not working

**Comment by hakonanes** [comment]
Hi @mjp-ua,

which specific file formats are you talking about? Oxford's binary .ebsp file to EDAX' binary .up1 or .up2 file?

We have readers for both these formats in kikuchipy ([supported formats](https://kikuchipy.org/en/stable/tutorials/load_save_data.html#Supported-file-formats)). kikuchipy extends HyperSpy for EBSD pattern analysis. But, we can only write to the NORDIF binary .dat format and our own custom HDF5 format (readable by EMsoft). Unless your OIM software can import NORDIF's binary file format without the corresponding metadata returned by NORDIF (settings, background pattern, calibration patterns, SE image with region of interest, etc.), I'm afraid kikuchipy is of little help.

PyEBSDIndex ([source](https://github.com/USNavalResearchLaboratory/PyEBSDIndex)) is another EBSD software that may do what you want. I know it has readers and writers for both binary formats. But, I'm not sure you can combine reading one format and writing to the other. @drowenhorst-nrl, is PyEBSDIndex able to do this?

**Reply by MJPavel1** [reply]
Thank you for the recommendation I will look into PyEBSDIndex. That makes sense, unfortunately, I think we would definitely run into problems on the import side but I appreciate the input.

**Comment by drowenhorst-nrl** [comment]
Hey @mjp-ua - 
While PyEBSDIndex does not directly have translation tools for the pattern formats, there are enough features of reading/writing various pattern files that it is not too hard to chain something together (and I have done this in the past).  Let me know what your current processing chain would look like, I might have a notebook that would work for you.  

It depends on what features of OIMAnalysis you are looking at (just the analysis, or also some parts of the indexing), but PyEBSDIndex will take in patterns from Oxford (certain .ebsp files are supported, the HDF5 format is much more easily supported), and perform the indexing. PyEBSDIndex does have an ang writer for the indexed data.  It also has a .oh5 writer (EDAX's HDF5 replacement for .ang), which the current release version has a bug, but I should have a patch out soon to fix that.  If you are impatient, my personal [develop branch](https://github.com/drowenhorst-nrl/PyEBSDIndex/tree/develop)  should have the fix.  We have a paper that is just about to come out that shows how PyEBSDIndex performs relative to the EDAX indexing - [PyEBSDIndexPaper](https://doi.org/10.1107/S1600576723010221)

**Reply by MJPavel1** [reply]
Hello that sounds exactly what I'd be looking for. In the past we were able to take our Aztec output files and open them in an old OIM version which we no longer have access to, save as an OIM file and then open in the new version.

To be honest I'm not sold on EDAX as whole - their analysis and viewpoint always strikes me as "If you have crappy data we can fix it" rather than high quality data first. That being said - we would love to try out NPAR - spherical / dictionary indexing for some materials with a lot of deformation. Otherwise, OIM is mostly useful for us for making figures. We only have an old copy of Tango for processing the Oxford data and it's just a little lackluster in it's data overlaying and partitioning capabilities.

I'll update this comment later today with what file formats we are actually exporting. I appreciate the help!

**Reply by jlaehne** [reply]
Kikuchipy has a function [Average neighbour patterns](https://kikuchipy.org/en/stable/reference/generated/kikuchipy.signals.EBSD.average_neighbour_patterns.html), which you might want to try out as well then.

**Reply by drowenhorst-nrl** [reply]
And not to be too much of a "me too", but PyEBSDIndex has an implementation of NLPAR, which is a slightly smarter, but slower method of pattern averaging (but this implemenation is much much faster than EDAX).  

Funny - my impressions have been quite opposite with EDAX, but it might originating from the same phenomena.  I have found that EDAX handles low quality data (low quality patterns) much better than "the others" - providing >99% hit rates, while oxford struggled to get 90%, thus people tend to collect lower quality patterns on EDAX. Maybe my impressions are dated, but I always thought Oxford the ones wanting to clean up their 5-10% of points that were not indexed. Nearly impossible to do apples-to-apples comparisons though.  Different SEMs, different cameras, different indexing ... 

Shoot me an email at david.rowenhorst at nrl dot navy dot mil 
and I will see if I can put together a notebook for conversion.
